### PR TITLE
programs.fish: add useBabelfish option

### PR DESCRIPTION
**Pull request by elohmeier**
### Description

Added useBabelfish option to programs.fish. Speeds up fish startup noticeably. Similar implementation as in NixOS fish module.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by emilazy** [comment]
See also: https://github.com/nix-community/home-manager/pull/4012

Sorry I didn't see this before writing mine...

File: modules/programs/fish.nix [hunk 1]
```diff
@@ -354,9 +374,15 @@ in {
         set -q __fish_home_manager_config_sourced; and exit
         set -g __fish_home_manager_config_sourced 1
 
-        set --prepend fish_function_path ${pkgs.fishPlugins.foreign-env}/share/fish/vendor_functions.d
-        fenv source ${config.home.profileDirectory}/etc/profile.d/hm-session-vars.sh > /dev/null
-        set -e fish_function_path[1]
+        ${if cfg.useBabelfish then ''
+          source ${
+            babelfishTranslate hm-session-vars "hm-session-vars"
+          } > /dev/null
```
Comment (ncfavier) [inline modules/programs/fish.nix:380]: Why `> /dev/null`?

**Review by ncfavier — COMMENTED** [review]
Oops, sorry I didn't see this PR!

**Comment by ncfavier** [comment]
Superseded by https://github.com/nix-community/home-manager/pull/4012

**Pull request closed without merging**